sdk/tables/Azure.Data.Tables/src/Extensions/DictionaryTableExtensions.cs

```diff
@@ -132,20 +132,19 @@ internal static List<T> ToTableEntityList<T>(this IReadOnlyList<IDictionary<stri
         /// <summary>
         /// Cleans a Dictionary of its Odata type annotations, while using them to cast its entities accordingly.
         /// </summary>
-        internal static T ToTableEntity<T>(this IDictionary<string, object> entity, TablesTypeBinder.BoundTypeInfo? typeInfo = null) where T : class, ITableEntity, new()
+        internal static T ToTableEntity<T>(this IDictionary<string, object> entity, TablesTypeBinder.BoundTypeInfo? typeInfo = null) where T : class, ITableEntity
         {
             if (typeof(IDictionary<string, object>).IsAssignableFrom(typeof(T)))
             {
-                var result = new T();
-                var dictionary = (IDictionary<string, object>)result;
+                var tableEntity = new TableEntity();
                 entity.CastAndRemoveAnnotations();
 
                 foreach (var entProperty in entity.Keys)
                 {
-                    dictionary[entProperty] = entity[entProperty];
+                    tableEntity[entProperty] = entity[entProperty];
                 }
 
-                return result;
+                return (T)(object)tableEntity;
```

Comment:
Will this always be a safe cast?   I may not be following fully, but it looks like we're casting from `TableEntity` to a derived type, which I wouldn't expect to work.

Reply:
Actually, you are right. Initially I was thinking that these would be either standard custom models or `TableEntity`. We need to preserve the type we got in for cases where a custom model type also implements `IDictionary<string, object>`